### ISSUE TITLE
`opening_brace` and `closing_brace` should not validate function arguments.

### ISSUE THREAD

**Issue by lhunath** [realm/SwiftLint#3690]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The goal of the `opening_brace` and `closing_brace` rules is to validate that braces are placed correctly on the same line and separated by the preceding syntax with a single space:

> Opening braces should be preceded by a single space and on the same line as the declaration.

Some parts of this rule exceed the scope of the rule and validate brace placement inside function arguments as having no spaces at all:

```swift
[].map({ })
```

This validation goes against the spirit of the rule and exceeds the its scope of responsibility, seeing as it's intent is to validate the syntax of function bodies.  And while it can be argued that both a trailing closure and an inline closure are both function bodies, the reality is that this validation performs the opposite of what it claims (yielding confusing validation failure messages) and has leaked into the territory of validating the syntax for function arguments.

Ideally, there should be a separate rule that validates the spacing used for function arguments and this rule should limit its scope to that described by its premise.  Pending a separate rule being introduced, this rule should be limited in scope regardless.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.43.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 12.5.1 Build version 12E507
